toolchain/semantics/semantics_handle_function.cpp

```diff
@@ -18,17 +18,10 @@ auto SemanticsHandleFunctionDefinition(SemanticsContext& context,
          ParseNodeKind::FunctionDefinitionStart) {
     context.node_stack().PopAndIgnore();
   }
-  auto function_id = context.node_stack().Pop<SemanticsFunctionId>(
-      ParseNodeKind::FunctionDefinitionStart);
-
+  context.node_stack().PopAndIgnore();
```

Comment:
Please strongly prefer PopAndDiscard over PopAndIgnore. The difference is that PopAndDiscard continues to validate structure, whereas PopAndIgnore doesn't -- allowing incorrect structure expectations to go on longer.
```suggestion
  context.node_stack().PopAndDiscardSoloParseNode(ParseNodeKind::FunctionDefinitionStart);
```